rotkehlchen/tests/data_migrations/test_migrations.py

```diff
@@ -384,39 +443,45 @@ def test_migration_11(
-                (UpdateType.SPAM_ASSETS.serialize(), 999),
-                (UpdateType.RPC_NODES.serialize(), 999),
-                (UpdateType.CONTRACTS.serialize(), 999),
-                (UpdateType.GLOBAL_ADDRESSBOOK.serialize(), 999),
-            ],
-        )
-    migration_patch = patch(
-        'rotkehlchen.data_migrations.manager.MIGRATION_LIST',
-        new=[MIGRATION_LIST[5]],
+    detect_accounts_migration_check(
+        chains=[SupportedBlockchain.ARBITRUM_ONE],
+        migration_version=11,
+        migration_steps=5,  # 2 (eth accounts) + 3 (potentially write to db + updating spam assets and rpc nodes + step 0)  # noqa: E501
+        migration_list=[MIGRATION_LIST[5]],
+        ethereum_accounts=ethereum_accounts,
+        rotkehlchen_api_server=rotkehlchen_api_server,
+        websocket_connection=websocket_connection,
     )
 
-    arbitrum_one_addresses = [ethereum_accounts[1]]
-    with migration_patch:
-        DataMigrationManager(rotki).maybe_migrate_data()
 
-    # check that detecting arbitrum one accounts works properly
-    with rotki.data.db.conn.read_ctx() as cursor:  # make sure DB is also written
-        accounts = rotki.data.db.get_blockchain_accounts(cursor)
-
-    assert set(accounts.eth) == set(ethereum_accounts)
-    assert set(rotki.chains_aggregator.accounts.eth) == set(ethereum_accounts)
-    assert set(accounts.arbitrum_one) == set(arbitrum_one_addresses)
-    assert set(rotki.chains_aggregator.accounts.arbitrum_one) == set(arbitrum_one_addresses)
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('data_migration_version', [11])
+@pytest.mark.parametrize('perform_upgrades_at_unlock', [True])
+@pytest.mark.parametrize('ethereum_accounts', [[
+    '0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84',  # mainnet contract
+    '0x78C13393Aee675DD7ED07ce992210750D1F5dB88',  # mainnet + gnosis + base + other evm chains activity  # noqa: E501
+]])
+@pytest.mark.parametrize('legacy_messages_via_websockets', [True])
+@pytest.mark.parametrize('network_mocking', [False])
+def test_migration_12(
+        rotkehlchen_api_server: 'APIServer',
+        ethereum_accounts: list[ChecksumEvmAddress],
+        websocket_connection: 'WebsocketReader',
+) -> None:
+    """
+    Test migration 12.
 
-    migration_steps = 5  # 2 (eth accounts) + 3 (potentially write to db + updating spam assets and rpc nodes + step 0)  # noqa: E501
-    chain_to_added_address = [{'evm_chain': 'arbitrum_one', 'address': ethereum_accounts[1]}]
-    assert_add_addresses_migration_ws_messages(websocket_connection, 11, migration_steps, chain_to_added_address)  # noqa: E501
+    - Test that detecting gnosis accounts works properly
+    """
+    detect_accounts_migration_check(
+        chains=[SupportedBlockchain.GNOSIS, SupportedBlockchain.BASE],
+        migration_version=12,
+        migration_steps=5,  # 2 (eth accounts) + 3 (potentially write to db + updating spam assets and rpc nodes + step 0)  # noqa: E501
```

Comment:
As I wrote in the migration it's weird to have same number of steps as migration 11 which only migrates 1 chain. Something is wrong.

Reply:
https://github.com/rotki/rotki/pull/6706#discussion_r1350319561 
The number of messages doesn't depend on the number of migrating chains. It only creates one message and adds the info about the chains in the message data. This data is checked in the tests.